benchmarks/bvh_driver/benchmarks_bvh.cpp

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <benchmark_registration.hpp>
+
+void register_bvh_benchmarks(Spec const &spec)
+{
+#ifdef KOKKOS_ENABLE_SERIAL
+  if (spec.backends == "all" || spec.backends == "serial")
+    register_benchmark<Kokkos::Serial,
+                       ArborX::BVH<typename Kokkos::Serial::memory_space>>(
+        "ArborX::BVH<Serial>", spec);
+#else
+  if (spec.backends == "serial")
+    throw std::runtime_error("Serial backend not available!");
+#endif
+
+#ifdef KOKKOS_ENABLE_OPENMP
+  if (spec.backends == "all" || spec.backends == "openmp")
+    register_benchmark<Kokkos::OpenMP,
+                       ArborX::BVH<typename Kokkos::OpenMP::memory_space>>(
+        "ArborX::BVH<OpenMP>", spec);
+#else
+  if (spec.backends == "openmp")
+    throw std::runtime_error("OpenMP backend not available!");
+#endif
+
+#ifdef KOKKOS_ENABLE_THREADS
+  if (spec.backends == "all" || spec.backends == "threads")
+    register_benchmark<Kokkos::Threads,
+                       ArborX::BVH<typename Kokkos::Threads::memory_space>>(
+        "ArborX::BVH<Threads>", spec);
+#else
+  if (spec.backends == "threads")
+    throw std::runtime_error("Threads backend not available!");
+#endif
+
+#ifdef KOKKOS_ENABLE_CUDA
+  if (spec.backends == "all" || spec.backends == "cuda")
+    register_benchmark<Kokkos::Cuda,
+                       ArborX::BVH<typename Kokkos::Cuda::memory_space>>(
+        "ArborX::BVH<Cuda>", spec);
```

Comment:
I had not noticed that before.  It is kinda sketchy.  Can be UVM depending on how Kokkos was built.

Reply:
Or was it the intent?

Reply:
I'm not sure. This was introduced a long time ago, and I'm unsure of the intent at that time. Not addressing this in this PR.